### PR TITLE
Update plugin ID of hub extension

### PR DESCRIPTION
**Pull request by mctoohey**
## References

Fixes https://github.com/jupyterlab/jupyterlab/issues/13687

## Code changes

Updates the plugin IDs of the `hub-extension` plugin to fix an issue where settings that add items to the file menu were no longer  being loaded.

## User-facing changes

None, other than the reappearance of the "Hub Control Panel" and "Log Out" File menu items for users using Jupyterlab with Jupyterhub.

## Backwards-incompatible changes

None that I am aware of. But someone with more knowledge and experience may be able to comment on any potential impact of changing the plugin ID.

### CODE REVIEW

File: packages/hub-extension/src/index.ts [hunk 1]
```diff
@@ -100,7 +100,7 @@ function activateHubExtension(
  */
 const hubExtension: JupyterFrontEndPlugin<void> = {
   activate: activateHubExtension,
-  id: 'jupyter.extensions.hub-extension',
+  id: '@jupyterlab/hub-extension',
```
Comment (jtpio) [inline packages/hub-extension/src/index.ts:103]: Maybe we could rename this plugin to `@jupyterlab/hub-extension:plugin` directly?

File: packages/hub-extension/src/index.ts [hunk 2]
```diff
@@ -111,7 +111,7 @@ const hubExtension: JupyterFrontEndPlugin<void> = {
  */
 const hubExtensionMenu: JupyterFrontEndPlugin<void> = {
   activate: () => void 0,
-  id: 'jupyter.extensions.hub-extension:plugin',
+  id: '@jupyterlab/hub-extension:plugin',
```
Comment (jtpio) [inline packages/hub-extension/src/index.ts:114]: And this one to `@jupyterlab/hub-extension:menu`?

**Review by jtpio — COMMENTED** [review]
Thanks!

Maybe there could be a quick mention of the plugin rename in the extension migration guide as well?

https://jupyterlab.readthedocs.io/en/latest/extension/extension_migration.html#api-breaking-changes

**Comment by mctoohey** [comment]
Thanks for the review and suggestions @jtpio. I've pushed a couple more commits to address your comments.

**Review by jtpio — APPROVED** [review]
Looks good, thanks @mctoohey!

**Pull request merged**